phpBB/phpbb/mention/helper/mention_helper.php

```diff
@@ -0,0 +1,196 @@
+use phpbb\controller\helper;
+
+class mention_helper
+{
+    /**
+    * @var driver_interface
+    */
+    private $db;
+
+    /**
+    * @var Notification Details
+    */
+    private $data;
+
+    /**
+    * Mention Helper Class Constructor
+    *
+    * @param $db   Database Driver Interface Object.
+    *
+    * @return \phpbb\mention\helper\mention_helper
+    */
+    public function __construct(driver_interface $db)
+    {
+        $this->db = $db;
+    }
+
+    /**
+    * Function to extract the mentioned users using regex exp.
+    *
+    * @param string    $regular_expression_match     regex exp.
+    * @param string    $post_text                    Text pilled up from the board.
+    *
+    * @return array Consisiting of matches.
+    */
+    public function get_regex_match($regular_expression_match, $post_text)
+    {
+        $matches = false;
+        preg_match_all($regular_expression_match, $post_text, $matches, PREG_OFFSET_CAPTURE);
+        return $matches;
+    }
+
+    /**
+    * Function to extract the mentioned users using regex exp.
+    *
+    * @param array    $matches   all matches with [mention][/mention].
+    *
+    * @return array List of all different users tagged in the text..
+    */
+    public function get_user_list($matches)
+    {
+        $user_list = array();
+        // get list of all different users tagged.
+        for ($i = 0, $len = count($matches[1]); $i < $len; $i++)
+        {
+            if (!in_array($matches[1][$i][0], $user_list, true))
+            {
+                array_push($user_list, $matches[1][$i][0]);
+            }
+        }
+        return $user_list;
+    }
+
+    /**
+    * Function to extract the mentioned users using regex exp.
+    *
+    * @param array    $matches            all the matches with regexp.
+    * @param integer  $start_tag_length   Length of tag [mention].
+    * @param integer  $end_tag_length     Length of tag [\mention].
+    * @param integer  $userid_list        Ids of users tagged in the post.
+    *
+    * @return array Consisiting of new text and array of userid.
+    */
+    public function get_regex_substituted_text($matches, $post_text, $start_tag_length, $end_tag_length, $userid_list)
+    {
+        $users_already_mapped = array();
+        //Iterate over all [mention][/mention] matches.
+        for ($i = 0, $len = count($matches[1]); $i < $len; $i++)
+        {
+            $username_clean = utf8_clean_string($matches[1][$i][0]);
+            //Inititalize all variables like length of usernames and link to all users.
+            $startpos = $matches[1][$i][1] - $start_tag_length;
+            $length = strlen($matches[1][$i][0]);
+            $endpos = $matches[1][$i][1] + $length + $end_tag_length - 1;
+            $userid = $userid_list[$username_clean];
+            $add_url_tag = "[url=" . generate_board_url() . "/memberlist.php?mode=viewprofile&u=" . $userid . "]" . $username_clean . "[/url]";
```

Comment:
I think we have a function to generate the link to the profile of a user. But I may be wrong